### PR TITLE
Fix `_TZ3000_zl1kmjqx` with empty model string

### PR DESCRIPTION
**Pull request by jpizquierdo**
## Proposed change

There are some _TZ3000_zl1kmjqx from aliexpress that has not specified a model, they are empty, so they only exposes temperature instead of temperature + humidty. I googled and faced that someone was having the same issue:
https://community.home-assistant.io/t/tz3000-zl1kmjqx-doesnt-show-humidity-entities/708283/3 so I thought that it could be fixed with a custom zhaquirk, I tested it in my custom quirk folder with this configuration inside configuration.yaml:
```
zha:
  database_path: /config/zigbee.db
  enable_quirks: true
  custom_quirks_path: /config/zha_quirks/
```

## Additional information
This is the previous device signature with "class": "zigpy.device.Device" instead of ty0201.TuyaTempHumiditySensor:
EDD646A5-5D33-4035-A4DD-E0BE55B4E399

This is the device signature fixed, as you can see model is empty and this was causing that the already implemented custom quirk was not working properly: 
```
{
  "node_descriptor": "NodeDescriptor(logical_type=<LogicalType.EndDevice: 2>, complex_descriptor_available=0, user_descriptor_available=0, reserved=0, aps_flags=0, frequency_band=<FrequencyBand.Freq2400MHz: 8>, mac_capability_flags=<MACCapabilityFlags.AllocateAddress: 128>, manufacturer_code=4417, maximum_buffer_size=66, maximum_incoming_transfer_size=66, server_mask=10752, maximum_outgoing_transfer_size=66, descriptor_capability_field=<DescriptorCapability.NONE: 0>, *allocate_address=True, *is_alternate_pan_coordinator=False, *is_coordinator=False, *is_end_device=True, *is_full_function_device=False, *is_mains_powered=False, *is_receiver_on_when_idle=False, *is_router=False, *is_security_capable=False)",
  "endpoints": {
    "1": {
      "profile_id": "0x0104",
      "device_type": "0x0302",
      "input_clusters": [
        "0x0000",
        "0x0001",
        "0x0003",
        "0x0402",
        "0x0405"
      ],
      "output_clusters": [
        "0x0019"
      ]
    }
  },
  "manufacturer": "_TZ3000_zl1kmjqx",
  "model": "",
  "class": "ty0201.TuyaTempHumiditySensor"
}
```

This is the sensor working with this change:
image

## Checklist

- [x] The changes are tested and work correctly
- [x] `pre-commit` checks pass / the code has been formatted using Black
- [x] Tests have been added to verify that the new code works

### CODE REVIEW

**Comment by jpizquierdo** [comment]
I updated the comment with an image showing that "class": "zigpy.device.Device" was used instead of ty0201.TuyaTempHumiditySensor

**Review by TheJulianJES — DISMISSED** [review]
Interesting. Looks like it's a device/firmware issue. Z2M seems to be doing the same.
This should be fine, thanks!

**Comment by TheJulianJES** [comment]
Hmm, the signature test fails. We might have to exclude that device from it.

**Comment by jpizquierdo** [comment]
Yeah probably, should I do something more? @TheJulianJES

**Comment by matthewmcneill** [comment]
I had this issue today after purchasing the devices from Amazon and adding them to a new Zigbee skyconnect.

I have attached a grep of my logs - grepping 'zha' if that helps.  It looks like some error at line 155:

2024-08-29 17:54:30.810 DEBUG (MainThread) [zha.zigbee.cluster_handlers] [0x8D2B:1:0x0405]: Failed to configure reporting for '['measured_value']' on 'humidity' cluster: [ConfigureReportingResponseRecord(status=<Status.UNSUPPORTED_ATTRIBUTE: 134>, direction=<ReportingDirection.SendReports: 0>, attrid=0)]

Not sure if that helps debug this at all - new to zigbee and going down the rabbit hole to understand quirks.

Matt

[zigbee_tuya_ty0201.log](https://github.com/user-attachments/files/16820707/zigbee_tuya_ty0201.log)

**Comment by jpizquierdo** [comment]
what can we do to make it work and merge?

**Comment by jpizquierdo** [comment]
@TheJulianJES Hello, could you check the PR? I have implemented some logic to have a white list for "manufacturers" to be able to have devices with an empty model and pass the tests

**Comment by TheJulianJES** [comment]
Can you do it similar to how the fixture for the `test_signature_model_info_given` test does it and just exclude the quirk class?

**Comment by jpizquierdo** [comment]
yeah, done. It was a better idea and way of doing it. Thanks

File: tests/test_quirks.py [hunk 1]
```diff
@@ -305,7 +305,18 @@ def test_quirk_quickinit(quirk: zigpy.quirks.CustomDevice) -> None:
...
 
-@pytest.mark.parametrize("quirk", ALL_QUIRK_CLASSES)
+@pytest.mark.parametrize(
+    "quirk",
+    [
+        quirk_cls
+        for quirk_cls in ALL_QUIRK_CLASSES
+        if quirk_cls
+        not in (
+            # Some devices do not have model info:
```
Comment (TheJulianJES) [inline tests/test_quirks.py]:
```suggestion
            # Some devices have empty model info:
```

**Review by TheJulianJES — APPROVED** [review]
Thanks, this looks good!

**Comment by JoopR-ops** [comment]
Can some-one please help me? I've got the TZ3000_zl1kmjqx version of the sensor. And i think I've added the quirk correctly but i still cannot get it to work. 

Schermafbeelding 2024-12-28 004402

**Pull request merged**